src/main/java/org/openstreetmap/atlas/utilities/filters/AtlasEntityPolygonsFilter.java

```diff
@@ -0,0 +1,520 @@
+                            polygons.add((Polygon) propertiesLocated.getItem());
+                        }
+                        else
+                        {
+                            logger.warn("Polygon Filter does not support item {}",
+                                    propertiesLocated.toString());
+                        }
+                    }
+                    return Optional.of(polygons).filter(list -> !list.isEmpty());
+                };
+            case WKT:
+                return string -> Optional.of(Collections
+                        .singletonList(new WktPolygonConverter().backwardConvert(string)));
+            case WKB:
+                return string -> Optional.of(Collections.singletonList(
+                        new WkbPolygonConverter().backwardConvert(WKBReader.hexToBytes(string))));
+            case UNSUPPORTED:
+            default:
+                logger.warn("No converter set up for {} format. Supported formats are {}", format,
+                        Arrays.copyOf(PolygonStringFormat.values(),
+                                PolygonStringFormat.values().length - 1));
+                return string -> Optional.empty();
+        }
+    }
+
+    private static List<Polygon> getPolygonsFromFormatMap(
+            final Map<String, List<String>> polygonLists)
+    {
+        if (polygonLists.isEmpty())
+        {
+            return Collections.emptyList();
+        }
+        return polygonLists.entrySet().stream()
+                .flatMap(formatAndPolygonStrings -> formatAndPolygonStrings.getValue().stream().map(
+                        getPolygonConverterForFormat(formatAndPolygonStrings.getKey())::convert)
+                        .filter(Optional::isPresent).map(Optional::get).flatMap(List::stream))
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Returns predicate that filters out MultiPolygons that overlap with any Polygons or outer
+     * MultiPolygons in the provided QuadTree. This also adds any non-intersecting MultiPolygon to
+     * the quad tree.
+     *
+     * @param vettedPolygons
+     *            non-overlapping polygons and multipolygons
+     * @param polygonType
+     *            whether this is an include or exclude Polygon
+     * @return filter for multipolygons
+     */
+    private static Predicate<MultiPolygon> notOverlappingMultipolygon(
+            final QuadTree<Located> vettedPolygons, final Type polygonType)
+    {
+        return multiPolygon ->
+        {
+            if (vettedPolygons.get(multiPolygon.bounds()).stream()
+                    .filter(located -> located instanceof Polygon).map(located -> (Polygon) located)
+                    .noneMatch(multiPolygon::overlaps)
+                    && vettedPolygons.get(multiPolygon.bounds()).stream()
+                            .filter(located -> located instanceof MultiPolygon)
+                            .map(located -> (MultiPolygon) located).map(MultiPolygon::outers)
```

Comment:
Do we care about the inners here?

Reply:
In this case no, because for a multi polygon to intersect with another multi polygon, the outers have to overlap with the other Multipolygon (inners and outs considered), and visa versa.

Reply:
That being said, not sure I did the visa versa part. So I'm going to make a function to do that.